packages/twenty-front/src/pages/settings/data-model/SettingsObjects.tsx

```diff
@@ -29,7 +39,25 @@ import { Table } from '@/ui/layout/table/components/Table';
 import { TableHeader } from '@/ui/layout/table/components/TableHeader';
 import { TableSection } from '@/ui/layout/table/components/TableSection';
 import { UndecoratedLink } from '@/ui/navigation/link/components/UndecoratedLink';
-
+import { useSortedArray } from '~/hooks/useSortedArray';
+export enum SortKeys {
+  labelPlural = 'labelPlural',
+  objectTypeLabelText = 'objectTypeLabelText',
+  fieldsCount = 'fieldsCount',
+  instancesCount = 'instancesCount',
+}
+type InstanceCountStateType = { [key: string]: number };
+type TableHeading = {
```

Comment:
The developer experience you're creating for future sort use-cases isn't great, you have to redeclare types every time you want to add sort to a table. You're already duplicating some code between SettingsObjects and SettingsObjectDetail